EpikCord/__init__.py

```diff
@@ -1216,6 +1216,44 @@ def option_autocomplete(self, option_name: str):
         def wrapper(func):
             self.autocomplete_options[option_name] = func
         return wrapper
+    def checks(self, *args):
+      
+        """Checks the conditions given to it
+        Parameters:
+            All parameters should be functions returning boolean
+        
+        Returns:
+            A wrapper for a function
+        """
+        
+        async def wrap(func:Callable):
+            check_no = len(args)
+            success_check = 0
+            for checker in args:
+                
+                checker:Callable = checker
+                if not asyncio.iscoroutinefunction(checker):
+                    checked:bool = checker()
+                    if checked:
+                        success_check += 1
+                        logger.debug(f"Checker {checker.__name__} for {func.__name__} is successful {success_check}/{check_no} PASS")
+                    else:
+                        logger.debug(f"Checker {checker.__name__} for {func.__name__} failed. Aborting.")
+                        break #No logger messages can be sent now haha!
+                else:
+                    checked:bool = await checker()
```

Comment:
And also support for normal functions won't hurt aswell i guess, but that is up to u.

Reply:
i did
look up

Reply:
oh i didnt notice that, :+1: